### PR TITLE
MAINT: SciPy 1.7.2 prep/backports

### PR DESCRIPTION
**Pull request by tylerjereddy**
With Python 3.10 coming out early next week, backport PRs with the appropriate label, update release notes, and see where we are with the CI, etc.

Backports included so far:
- gh-14207
- gh-14316
- gh-14323
- gh-14478
- gh-14618
- gh-14672
- gh-14333
- gh-14605

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -18,10 +18,10 @@ pr:
     include:
     - '*'  # must quote since "*" is a YAML reserved character; we want a string
 
-# the version of OpenBLAS used is currently 0.3.8.dev
+# the version of OpenBLAS used is currently 0.3.15
```
Comment (tylerjereddy) [inline azure-pipelines.yml:21]: comment looks like version typo, didn't check if still on master, not super important for now

File: doc/release/1.7.2-notes.rst [hunk 2]
```diff
@@ -5,16 +5,39 @@ SciPy 1.7.2 Release Notes
...
+version of OpenBLAS, 0.3.17.
 
 Authors
 =======
 
+* Ralf Gommers
+* Matt Haberland
+* Nicholas McKibben
+* Judah Rand +
+* Tyler Reddy
```
Comment (tylerjereddy) [inline doc/release/1.7.2-notes.rst:19]: Folks are helping drive some wheels repo changes--that should probably be taken into account in this list as well.

File: azure-pipelines.yml (hunk 1 above)
Comment (rgommers) [reply azure-pipelines.yml:21]: yes, I believe that should be `17`

**Review by rgommers — COMMENTED** [review]
What is here looks good to backport. The OpenBLAS support may also need a backport of gh-14605, and gh-14333 could be relevant as well.

**Comment by rgommers** [comment]
The 32-bit Linux build now ends in:
```
numpy.distutils.system_info.NotFoundError: No BLAS/LAPACK libraries found.
```
gh-14333 may indeed resolve that

**Comment by tylerjereddy** [comment]
Updated with the two additional backports, release notes with those backports and more inclusive author list, and typo fix in azure config comment.

Note gh-14605 had pretty substantial merge conflicts--I tried to use caution, but we'll see how it goes.

**Comment by tylerjereddy** [comment]
the remaining test failure is gh-14732; we've just been busy I think--that one still exists on `master` as well

**Review by rgommers — APPROVED** [review]
> Note [gh-14605](https://github.com/scipy/scipy/pull/14605) had pretty substantial merge conflicts--I tried to use caution, but we'll see how it goes.

Changes LGTM.



> the remaining test failure is [gh-14732](https://github.com/scipy/scipy/issues/14732); we've just been busy I think--that one still exists on `master` as well

Yep, let's ignore that one. Eventually we're going to regret not figuring that one out sooner, but I am swamped right now so 🤷🏼.

**Pull request merged**